cetera-http/src/main/scala/com/socrata/cetera/types/User.scala

```
@@ -60,15 +61,15 @@ object EsUser {
   implicit val codec = AutomaticJsonCodecBuilder[EsUser]
   val logger = LoggerFactory.getLogger(getClass)
 
-  def apply(source: String): Option[EsUser] =
-    Option(source).flatMap { s =>
+  def apply(source: String): EsUser =
+    Option(source).map { s =>
       JsonUtil.parseJson[EsUser](s) match {
```

I was curious if you had changed this and I'm further curious why you didn't just simplify to: 

```
JsonUtil.parseJson[EsUser](s) match {
  case Right(user) => user
  case Left(err) =>
    logger.error(err.english)
    throw new JsonDecodeException(err)	        
}
```

given that 

1. I thought you wanted that wrapping of nulls in `Option` to go away.
2. the exception thrown is then swallowed and turned into None anyway

---

@aynleslie thanks for pointing this out. I actually don't like this, but not for the reason you point out. Firstly, note that if we don't Option away the null, then we get a null source `s`, that won't be handled w/o an explicit try/catch.

```
scala> JsonUtil.parseJson[EsUser](null)
java.lang.NullPointerException
  at java.io.StringReader.<init>(StringReader.java:50)
  at com.rojoma.json.v3.io.FusedBlockJsonReader.<init>(FusedBlockJsonReader.scala:11)
  at com.rojoma.json.v3.io.JsonReader$.apply(JsonReader.scala:19)
  at com.rojoma.json.v3.io.JsonReader$.fromString(JsonReader.scala:41)
  at com.rojoma.json.v3.util.JsonUtil$.parseJson(JsonUtil.scala:61)
  ... 43 elided
```

It seems to me that this package should not be concerned w/ Elasticsearch errors. We should take a source string, assume it's not null (because we should never be passing nulls around), and therefore do away w/ the optioning. But that would mean that we have to add that null check in the appropriate places (presumably, the ES clients -- components that are concerned w/ Elasticsearch) and are aware of such quirks in the Elasticsearch API.